File: Domain/Ticket.cs
namespace Domain
{
    public class Ticket
    {
        public Guid Id { get; set; }
        public string Title { get; set; }

        public string Description { get; set; }
        public string Submitter { get; set; }
        public string Assigned { get; set; }
        public string Priority { get; set; }
        public string Severity { get; set; }
        public string Status { get; set; }
        public DateTime? ClosedDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime Updated { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public Guid ProjectId { get; set; }
        public Project Project { get; set; }
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: client-app/src/app/models/ticket.ts
import { Project } from "./project";

export interface Ticket {
  id: string;
  title: string;
  description: string;
  submitter: string;
  assigned: string;
  priority: string;
  severity: string;
  status: string;
  closedDate: Date | null;
  startDate: Date | null;
  endDate: Date | null;
  updated?: Date | null;
  createdAt?: Date | null;
  isDeleted?: boolean;
  deletedDate?: Date | null;
  projectId: string;
  projectTitle?: string;
  project?: Project;
}

export interface TicketFormValues {
  title: string;
  description: string;
  projectId: string;
}
